### DISCUSSION TITLE
Easy way to detect bpftrace has attached to all the probes and is ready to trace?

### DISCUSSION THREAD

**Discussion by 0mp** [bpftrace/bpftrace#2515] [Q&A]
Hello!

I'm dealing with large bpftrace scripts and sometimes it takes a couple of seconds or minutes for bpftrace to start tracing. This is not an issue if I'm providing the command via `-c`, but I'd like to know if there is a way detect that the bpftrace program is ready to trace in general. I'd like to run my command outside of bpftrace (i.e., without using `-c`) but I need to know that bpftrace is ready.

Now that I think about it, I could probably use `-c` for that, e.g., `-c "/bin/sh -c 'touch /tmp/bpftrace.ready && sleep infinity'"`. If there is a better way that does not feel like a hack, then please let me know.

**Comment by viktormalik** [comment]
This is an interesting point. I'd have to look more but I believe that the probes are attached in the order that they appear in the script. So, if you create a `BEGIN` probe as the last probe of you script and let it print something, then once the text is printed, the script *should* be ready.

**Reply by 0mp** [reply]
Hmm, that's an interesting implementation detail that could help. Thanks.
